WordPress/wporg-main-2022
# Fix export of content with list items

Incremental improvement to parsing list items. Currently if they contain other tags (eg. strong, em, a) the output markup in the resulting pattern file is broken. This PR copies the parser for paragraphs, changes the regex from `p` to `li`, and updates already exported patterns. 

Fixes #177 

Props @tellyworth , @StevenDufresne 

### How to test the changes in this Pull Request:

1. Add a page `/about-2/features-2/` to your site and the following to `page-manifest.json`
```
{
	"slug": "features-2",
	"pattern": "features.php",
	"template": "page-features.html"
},
```
and this to your `theme-switcher.php`
```
'/about-2/',
'/about-2/features-2/',
```
2. Run `yarn build:patterns`
3. Inspect the markup of the generated file `features.php` and observe that the markup in `patterns/features.php` is valid PHP, and text is escaped for translation. In particular check the output of `li` tags which include `a`, `strong`, `br`, tags. It should be consistent with the output of `p` tags.
4. For the page `/about-2/features-2/` choose the Features page template and run the page. There should be no fatal and the page should display like prod.
